packages/react-hooks/README.md

```diff
@@ -1,11 +1,301 @@
+
+	const square = useCallback(()=>{
+		setAnotherCounter(counter**2)
+	},[counter])
+
+	useDebounceEffect(square, 300, [counter])
+
+	return (
+		<>
+			<button onClick={() => setCounter(counter + 1)}>
+				Counter +1
+			</button>
+			<div>{counter}</div>
+			<div>{anotherCounter}</div>
+		</>
+	)
+}
+```
+>300 ms after the button is clicked, the callback function `square` is invoked, which squares the value of counter and assigns the result to the state of `anotherCounter`.
+## `useFileSelect(options)`
+`useFileSelect` selects the file(s) and returns a promise, which resolves with a `FileList`.
+#### Input
+```javascript
+options: {
+	multiple: boolean,
+	accept: string
+}
+```
+#### Output
+```javascript
+() => Promise<FileList>
+```
+#### Example
+```javascript
+import { useFileSelect } from "@w2solutions/react-hooks"
+
+const FileSelector = () => {
+	const  selectFiles = useFileSelect(
+		{multiple: false, accept: ".txt"}
+	)
+
+	const [fileName, setFileName] = useState<string | null>(null)
+
+	const callSelectFiles = useCallback(async () => {
+		const files = await selectFiles()
+		setFileName(files[0].name)
+	}, [selectFiles, setFileName])
+	
+	return (
+		<>
+			<button onClick={()=>callSelectFiles()}>
+				Select files
+			</button>
+			<div>Selected file: {fileName}</div>
+		</>
+	)
+}
+```
+> `useFileSelect` creates an asynchronous function `selectFiles`, which accepts 1 .txt file and resolves with `FileList`, which is stored in `files`. See for more info on [FileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList).
+## `useFormFields(initialValues, options)`
+`useFormFields` builds binding objects for input fields initialized with `initialValues`. Additionally the hook returns the current values of input fields, the boolean, which tells you if the form has been changed compared to the initial state, a reset function and a `setValue` function. `ResetOnInitialValueChange` boolean decides if the form should be reseted to new inital values (if they have been changed) and hence set the `isDirty` flag to false.
+#### Input
+```javascript
+initialValues = {
+	[key: string]: any
+}
+options = {
+	resetOnInitialValueChange?: boolean
+}
+```
+#### Output
+```javascript
+{
+	bindings: {
+		[key in keyof initialValues]: {
+			{
+				name: string,
+				value: initialValues[key],
+				onChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
+			}
+		}
+	},
+	values: {
+		[key in keyof initialValues]: initialValue[key]
+	},
+	isDirty: boolean,
+	reset: () => void,
+	setValue: (key, value) => void
+}
+```
+#### Example
+```javascript
+import { useFormFields } from "@w2solutions/react-hooks"
+
+const Form = () => {
+	const [color, setColor] = useState("red")
+	const { bindings, values, isDirty, reset, setValue } = useFormFields(
+		{color: color, shape: ''},
+		{resetOnInitialValueChange: true}
+	)
+
+	const submit = useCallback(
+		(event: React.FormEvent<HTMLFormElement>) => {
+			event.preventDefault()
```

Comment:
this example is a bite too complicated for this hook. i like it because it shows every feature.
though in general one does not need every feature. and there should be a simple example to show how to use it in general.

```tsx
const {bindings, values, isDirty, reset, setValue } = useFormFields({ firstName: '', lastName: '', age: 18 })

return (
  <form>
    <input {...bindings.firstName} type="text" />
    <input {...bindings.lastName} type="text" />

    <input
      type="number"
      onChange={(evt) => setValue('age', Number(evt.currentTarget.value))}
      {...bindings.age}
    /> 
   
    <button onClick={() => setValue('firstName', 'John')}>Set First Name</button>
    <button onClick={() => reset()}>Reset</button>
  <form>
)

```